silica/compile.py

```diff
@@ -81,16 +82,34 @@ def compile(coroutine, file_name=None, mux_strategy="one-hot", output='verilog',
     inline_yield_from_functions(tree, func_globals, func_locals)
     constant_fold(tree)
     specialize_list_comps(tree, func_globals, func_locals)
-    desugar_for_loops(tree)
+    tree, list_lens = propagate_types(tree)
+    tree, loopvars = desugar_for_loops(tree, list_lens)
+
+    ast_utils.print_ast(tree)
```

Comment:
Let's take this out before we merge. More generally it would be good to set up some logging infrastructure so we could have these print outs hidden under a `DEBUG` level. I'll put that on the todo list.